Improve "Material You" design

gcantoni opened the pull request:
Hello,

I reopen a pull request with the changes made in the previous one and other new ones.

This commit include several changes to the core and non-core layouts of SD Maid. At present the app does not implement a good part of the colors defined in colors within its layouts. For example, the main backgrounds and card backgrounds are not used correctly.

**Examples:**
- layout/dashboard_fragment.xml was missing attribute background
- setup_shizuku_item.xml  was missing both cardCornerRadius and cardBackgroundColor attributes
- many layouts do not correctly implement colors in the correct way. As said by you, the attributes should be inherited from the Material3 dependency. This is correct. The problem is that they don't use the official colors of Material Design 3. Just inspect some Google apps to quickly understand that behind their software there is something more in terms of UI compared to the native libraries. This requires the implementation of various additional attributes to achieve full design guidelines.

To answer your other question regarding toolbar colors, I simply restyled it using the correct parent and correct attributes. See https://github.com/gcantoni/sdmaid-se/blob/main/app/src/main/res/values/styles.xml#L194

Furthermore, to achieve full compatibility with Material You's dynamic colors I had to add the color of the status bar and navigation bar programmatically. See https://github.com/gcantoni/sdmaid-se/blob/main/app/src/main/java/eu/darken/sdmse/common/theming/Theming.kt#L130

It is logically all divided by API. For APIs > 34 the colors are the correct ones. (Refer to Google documentation here https://github.com/material-components/material-components-android/blob/master/docs/theming/Color.md)

Regarding API 31 (https://github.com/gcantoni/sdmaid-se/blob/main/app/src/main/java/eu/darken/sdmse/common/theming/Theming.kt#L147) feel free to change the colors as you like (see https://github.com/gcantoni/sdmaid-se/blob/main/app/src/main/java/eu/darken/sdmse/common/theming/Theming.kt#L157). Help yourself with the documentation that I have attached so you can understand how the colors of the Material 3 library are linked to the Android system resources.

**What needs to be improved / fixed?**
I can't find the background colors of these pages in the XMLs (I think you set them programmatically because even having added the attribute it doesn't change the background). Check it out, for you who know the app better than me it's a 2 minute thing.
[Screenshot_20250209-115220](https://github.com/user-attachments/assets/ae44327a-1e67-4049-bcb5-ce45436ea2c4)
[Screenshot_20250209-115224](https://github.com/user-attachments/assets/3fc88eb6-9583-4946-9c04-e39a2039fc4b)
[Screenshot_20250209-115228](https://github.com/user-attachments/assets/5bd7a43d-a7ef-4732-accf-d4b4cc3e31e6)

**Some screenshots**

- Basic Light Theme
[Screenshot_20250209-114933](https://github.com/user-attachments/assets/bb5fe835-a297-4d33-8ddd-36364e16626a)
[Screenshot_20250209-114937](https://github.com/user-attachments/assets/9707eedc-0b7f-4302-9820-e36550d63885)
[Screenshot_20250209-114947](https://github.com/user-attachments/assets/da055858-b43a-4925-9522-f2a41da02573)
[Screenshot_20250209-114958](https://github.com/user-attachments/assets/658158fb-c3b6-4201-a18b-f7ed688a5766)
[Screenshot_20250209-115010](https://github.com/user-attachments/assets/a193e302-f698-44c2-898a-03e7b6623fa1)

- Basic Dark Theme
[Screenshot_20250209-115035](https://github.com/user-attachments/assets/f6b0d619-a4dd-4dc4-a1c0-b367e00e728e)
[Screenshot_20250209-115037](https://github.com/user-attachments/assets/0a556224-e588-4475-a5d9-04c637070537)
[Screenshot_20250209-115048](https://github.com/user-attachments/assets/68d402ca-eb31-4ce0-ac8a-556f63ef6537)
[Screenshot_20250209-115053](https://github.com/user-attachments/assets/375976a5-381d-44d7-b9da-6457b9e91403)
[Screenshot_20250209-115103](https://github.com/user-attachments/assets/394d5bad-cd0a-4fd8-ae87-8b4e310c237c)

- Material You Light Theme

[Screenshot_20250209-115114](https://github.com/user-attachments/assets/dbac23e8-485c-4f3d-8826-d254c5e020f6)
[Screenshot_20250209-115117](https://github.com/user-attachments/assets/72e564ab-ce10-417c-bca0-84a47ac86946)
[Screenshot_20250209-115122](https://github.com/user-attachments/assets/477471a3-95f9-4009-9605-0076c0d11075)
[Screenshot_20250209-115127](https://github.com/user-attachments/assets/a001920a-365d-4a15-8da4-9f73bbf8df7d)
[Screenshot_20250209-115134](https://github.com/user-attachments/assets/c65eb2bc-d866-408f-aab1-fd5da7569c83)

gcantoni commented:
- Material You Dark Theme
![Screenshot_20250209-115144](https://github.com/user-attachments/assets/fc7314c0-e515-4df7-ab60-ae75118f5287)
![Screenshot_20250209-115147](https://github.com/user-attachments/assets/9f9f3cfc-3068-4e5d-b8be-691e38fc8306)
![Screenshot_20250209-115151](https://github.com/user-attachments/assets/aa740337-4d33-4f20-8461-db0b0baad758)
![Screenshot_20250209-115157](https://github.com/user-attachments/assets/197d28af-8c63-4ebd-a16c-b6c9ec70a8ec)
![Screenshot_20250209-115204](https://github.com/user-attachments/assets/1c958075-1771-4039-8913-2a6944b54920)

d4rken commented on app/local.properties:
```diff
@@ -0,0 +1,8 @@
+## This file must *NOT* be checked into Version Control Systems,
```
This file says it should not be checked into VCS :wink:

d4rken commented:
Thanks for the PR, will review it this week :heart: 

Ref #1469 #1579

gcantoni commented:
You are welcome Matthias. Feel free to drop me an email if you have any doubts. 

There would be other distinctive elements of Material Design 3 to implement, such as an expanding toolbar and some other small changes. However, these changes already integrate Material Design 3 very well, bringing SD Maid's UI to a decidedly higher level.

gcantoni commented:
@d4rken please include also this commit: https://github.com/gcantoni/sdmaid-se/commit/e320446b6bd711c50a32a082b2794afc4a23833a

1) I added padding to all the cards (example: https://github.com/gcantoni/sdmaid-se/commit/e320446b6bd711c50a32a082b2794afc4a23833a#diff-c22cff08a1c1a91e7b2aaf1b1c331262e7e599df9f8c7b9f0173b3cb0b84a8f2R27) 
2) I fixed a style that caused an unpleasant white color tone (https://github.com/gcantoni/sdmaid-se/commit/e320446b6bd711c50a32a082b2794afc4a23833a#diff-1e051f2b0330608e92eba8edc616403ec7bdaa76e29ce0cfb794700686715925L199) 
3) I replaced the material and non-compose switches in the planner to stay consistent with the rest of the switches (https://github.com/gcantoni/sdmaid-se/commit/e320446b6bd711c50a32a082b2794afc4a23833a#diff-009f36bb3edf7a78a1f63364442e1610a6b5d1404e23add7bd2733ec6da9e460L69)

d4rken commented:
> What needs to be improved / fixed?
I can't find the background colors of these pages in the XMLs (I think you set them programmatically because even having added the attribute it doesn't change the background). Check it out, for you who know the app better than me it's a 2 minute thing.
[Screenshot_20250209-115220](https://github.com/user-attachments/assets/ae44327a-1e67-4049-bcb5-ce45436ea2c4)
[Screenshot_20250209-115224](https://github.com/user-attachments/assets/3fc88eb6-9583-4946-9c04-e39a2039fc4b)
[Screenshot_20250209-115228](https://github.com/user-attachments/assets/5bd7a43d-a7ef-4732-accf-d4b4cc3e31e6)

Should be all set via XML. I don't remember any fragment programmatically setting the background, I'll look into it.

---

![Screenshot from 2025-02-11 17-03-13](https://github.com/user-attachments/assets/a8d92a77-384d-4cec-beec-50fe5a679151)

Should the navigation color here be "light" or "dark", white icons might be a bit difficult to see :thinking:, what do you think?

---

> I added padding to all the cards 

Need to weigh style/padding vs information density! Will look through it.

gcantoni commented:
The navigation bar must be dark. Please add to AppTheme style (https://github.com/d4rken-org/sdmaid-se/blob/main/app%2Fsrc%2Fmain%2Fres%2Fvalues%2Fthemes.xml#L69) the following attribute 

`<item name="android:windowLightNavigationBar" tools:targetApi="27">true</item>`

gcantoni commented:
> Is there a difference between `?attr/colorSurface` and `?colorSurface`?

Use always ?attr/colorSurface

d4rken commented:
> > Is there a difference between `?attr/colorSurface` and `?colorSurface`?
> 
> Use always ?attr/colorSurface

But why?

gcantoni commented:
> > The navigation bar must be dark. Please add to AppTheme style (https://github.com/d4rken-org/sdmaid-se/blob/main/app%2Fsrc%2Fmain%2Fres%2Fvalues%2Fthemes.xml#L69) the following attribute
> > 
> > `<item name="android:windowLightNavigationBar" tools:targetApi="27">true</item>`
> 
> Looks good :+1: thanks for the attribute
> 
> ![Screenshot from 2025-02-11 18-27-50](https://github.com/user-attachments/assets/e6b6e417-5f91-493b-94a2-ee2bd628b332)
> 
> ---
> 
> I like the color changes you made :heart: I will include most of them in the next build.
> 
> I'm not a fan of the very round cards (radius `28dp`) and will keep using the default card radius (`12dp`).
> 
> I'm current experimenting with the horizontal list padding. I agree with you that a bit more would be nice, not sure if `8dp`. Also i prefer to set the padding on the list content, not on the top-level lists, which prevents the viewholders from being design indepent.
> 
> Is the `8dp` padding something you picked based on what you like, or is this something that is specified by the MD docs. [I could only find that `8dp` is the padding between double column cards.](https://m3.material.io/components/cards/specs)
> 
> So that would be 4dp per card on the inside? :thinking: I'll see how it looks. 16dp also doesn't look too bad, but it really removes a lot of UI space for information, especially with some people increasing font-size.
> 
> Thanks for your help and feedback my friend :beers:
> 
> 
> 

You should use at least `16dp` for cards radius. 

Regarding the padding, I picked the one which looked better in my opinion. Google itself doesn't follow its guidelines so find the one which looks better for you.

You are welcome, Matthias.

gcantoni commented:
> > > Is there a difference between `?attr/colorSurface` and `?colorSurface`?
> > 
> > Use always ?attr/colorSurface
> 
> But why?
> 

Always use `?attr/colorSurface`. It ensures the value comes from the theme, making your Ul adaptive. `? colorSurface` is incorrect; use `@color/colorSurface` only if it's defined in colors.xml

Basically `?attr/example` ensures the value comes from the theme and can be overridden in different themes. Instead `?example` is an incorrect usage because is not a valid reference format.

d4rken reviewed (dismissed):
lgtm
